yt/fields/field_info_container.py

```diff
@@ -376,11 +376,13 @@ def load_all_plugins(self, ftype: Optional[str] = "gas"):
             only_on_root(mylog.debug, "Loaded %s (%s new fields)", n, len(loaded))
         self.find_dependencies(loaded)
 
-    def load_plugin(self, plugin_name, ftype="gas", skip_check=False):
-        if callable(plugin_name):
-            f = plugin_name
```

Comment:
Here's a small backward incompatible change I made to try and simplify type checking this function. It doesn't appear to break any tests so I assume that's ok, but if not, I can reintroduce (and possibly deprecate) this bit of logic.